src/test/java/StringManipulationTest.java

```diff
@@ -24,90 +24,159 @@ public void testCount1() {
-        fail("Not yet implemented");
+        manipulatedstring.setString("Hello");
+        IndexOutOfBoundsException testException = assertThrows(IndexOutOfBoundsException.class, () -> {
+        	manipulatedstring.removeNthCharacter(10, true);
+        });
+        assertEquals("Error: n exceeds string length", testException.getMessage());
     }
+ // this test checks whether method removeNthCharacter throws an IndexOutOfBoundsException if n exceeds string length
 
     @Test
     public void testRemoveNthCharacter4() {
-        fail("Not yet implemented");
+        manipulatedstring.setString("Testing123");
+        assertEquals("Tetig13", manipulatedstring.removeNthCharacter(3, false));
     }
+    
+ // this test checks whether method removeNthCharacter prints out the right string, which is "Tetig13"
 
     @Test
     public void testRemoveNthCharacter5() {
-        fail("Not yet implemented");
+        manipulatedstring.setString("Welcome to the jungle");
+        assertEquals("W l o e t   h   u g e", manipulatedstring.removeNthCharacter(2, true));
     }
+    
+    //this test checks whether method removeNthCharacter prints out the right string, which is "W l o e t   h   u g e"
 
     @Test
     public void testRemoveNthCharacter6() {
-        fail("Not yet implemented");
+        manipulatedstring.setString("I am loving it!");
+        IllegalArgumentException testException = assertThrows(IllegalArgumentException.class, () -> {
+        	manipulatedstring.removeNthCharacter(-54, true);
+        });
+        assertEquals("Error: n is less than or equal to 0", testException.getMessage());
     }
+    
+ // this test checks whether method removeNthCharacter throws an IllegalArgumentException if n is less than or equal to 0
 
     @Test
     public void testRemoveNthCharacter7() {
-        fail("Not yet implemented");
+        manipulatedstring.setString(null);
+        NullPointerException testException = assertThrows(NullPointerException.class, () -> {
+        	manipulatedstring.removeNthCharacter(5, true);
+        });
+        assertEquals("Error: String has not been initialized", testException.getMessage());
     }
+ // this test checks whether method removeNthCharacter throws an NullPointerException if string is null
 
     @Test
-    public void testGeSubStrings1() {
+    public void testGetSubStrings1() {
         manipulatedstring.setString("This is my string");
         String [] sStings = manipulatedstring.getSubStrings(3, 4);
 
         assertEquals(sStings[0], "my");
         assertEquals(sStings[1], "string");
     }
+ // this test checks whether method getSubStrings returns the right substring, which is "my" and "string"
 
     @Test
-    public void testGeSubStrings2() {
-        fail("Not yet implemented");
+    public void testGetSubStrings2() {
+    	manipulatedstring.setString(null);
+        NullPointerException testException = assertThrows(NullPointerException.class, () -> {
```

Comment:
I don't think you need a test for null pointer exception since this would fall under the other exceptions. It might be best to redo this test so you can test this under illegal argument exception